tests/csapi/apidoc_room_create_test.go

```diff
@@ -61,6 +61,72 @@ func TestRoomCreate(t *testing.T) {
 			})
 			content := alice.MustGetStateEventContent(t, roomID, "m.room.topic", "")
 			must.MatchGJSON(t, content, match.JSONKeyEqual("topic", "Test Room"))
+
+			// The plain text topic is duplicated into m.topic
+			must.MatchGJSON(t, content,
+				match.JSONKeyArrayOfSize("m\\.topic.m\\.text", 1),
+				match.JSONKeyPresent("m\\.topic.m\\.text.0.body"),
+				match.JSONKeyEqual("m\\.topic.m\\.text.0.body", "Test Room"))
+
+			// The mime type must be unset or text/plain
+			mime := content.Get("m\\.topic.m\\.text.0.mimetype")
+			if mime.Exists() {
+				must.Equal(t, mime.String(), "text/plain", "no m.topic content block")
+			}
+		})
+		// POST /createRoom makes a room with a topic via initial_state
+		t.Run("POST /createRoom makes a room with a topic via initial_state", func(t *testing.T) {
+			t.Parallel()
+
+			roomID := alice.MustCreateRoom(t, map[string]interface{}{
+				"initial_state": []map[string]interface{}{
+					{
+						"content": map[string]interface{}{
+							"topic": "Test Room",
+						},
+						"type":      "m.room.topic",
+						"state_key": "",
+					},
+				},
+				"preset": "public_chat",
+			})
+			content := alice.MustGetStateEventContent(t, roomID, "m.room.topic", "")
+			must.MatchGJSON(t, content, match.JSONKeyEqual("topic", "Test Room"))
+
+			// There is no m.topic property
+			must.MatchGJSON(t, content, match.JSONKeyMissing("m\\.topic"))
+		})
+		// POST /createRoom makes a room with a topic via initial_state overwritten by topic
+		t.Run("POST /createRoom makes a room with a topic via initial_state overwritten by topic", func(t *testing.T) {
```

Comment:
For my own reference, this behavior is spec'ed in the `/createRoom` endpoint, https://spec.matrix.org/v1.14/client-server-api/#post_matrixclientv3createroom


> The server MUST apply the normal state resolution rules when creating the new room, including checking power levels for each event. It MUST apply the events implied by the request in the following order:
>
> [...]
> 6. Events listed in `initial_state`, in the order that they are listed.
> 7. Events implied by `name` and `topic` (`m.room.name` and `m.room.topic` state events).
> [...]